### PR TITLE
minizip2: replace with minizip-ng

### PR DESCRIPTION
**Pull request by neheb**
Upstream changed the name. Rename the package accordingly.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Something seems wrong with this package. It seems all packages that depend on it use

```
${MINGW_PACKAGE_PREFIX}-minizip
```

but only minizip-git (which is quite old) provides it. Do I have that right?

### CODE REVIEW

File: mingw-w64-minizip-ng/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+_realname=minizip-ng
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}2")
+pkgver=3.0.4
+pkgrel=1
+pkgdesc="Minizip zlib contribution fork with latest bug fixes that supports PKWARE disk splitting, AES encryption, and IO buffering (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
+url='https://github.com/zlib-ng/minizip-ng'
+license=('ZLIB')
```
Comment (MehdiChinoune) [inline mingw-w64-minizip-ng/PKGBUILD:12]:
```suggestion
license=('ZLIB')
provides=("${MINGW_PACKAGE_PREFIX}-minizip2")
replaces=("${MINGW_PACKAGE_PREFIX}-minizip2")
conflicts=("${MINGW_PACKAGE_PREFIX}-minizip2")
```

Comment (neheb) [reply mingw-w64-minizip-ng/PKGBUILD:12]: OTOH nothing currently depends on minizip2.

Comment (MehdiChinoune) [reply mingw-w64-minizip-ng/PKGBUILD:12]: for people using minizip2 get updates!

**Comment by neheb** [comment]
@dennisameling can you test this?

File: mingw-w64-minizip-ng/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# Maintainer: Andrew Sun <adsun701@gmail.com>
+
+_realname=minizip-ng
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}2")
```
Comment (jeremyd2019) [inline mingw-w64-minizip-ng/PKGBUILD:5]:
```suggestion
pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
```

Or was this supposed to be `minizip-ng2`?

Comment (neheb) [reply mingw-w64-minizip-ng/PKGBUILD:5]: That 2 looks like a mistake. will remove.

Comment (neheb) [reply mingw-w64-minizip-ng/PKGBUILD]: Fixed.

**Comment by neheb** [comment]
So looking at the way fedora does this, it installs the includes to /include/minizip as is done with the original minizip (or minizip-git here). Unfortunately I can't see a way to do this without patching. This is normally done with the cmake flag: DINSTALL_INC_DIR , but for msys2, "DINSTALL_INC_DIR=${MINGW_PREFIX}/inlude/minizip" evaluates to

```
C:/msys64/home/Mangix/devstuff/MINGW-packages/mingw-w64-minizip-ng/pkg/mingw-w64-clang-i686-minizip-ng/msys64/clang32/inlude/minizip/mz.h
```

instead of the correct

```
C:/msys64/home/Mangix/devstuff/MINGW-packages/mingw-w64-minizip-ng/pkg/mingw-w64-clang-i686-minizip-ng/clang32/inlude/minizip/mz.h
```

**Comment by jeremyd2019** [comment]
You need the CONV_EXCL env var.

File: mingw-w64-minizip-ng/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  if check_option "debug" "n"; then
+    extra_config+=("-DCMAKE_BUILD_TYPE=Release")
+  else
+    extra_config+=("-DCMAKE_BUILD_TYPE=Debug")
+  fi
+
+  [[ -d "${srcdir}/build-${MSYSTEM}-shared" ]] && rm -rf "${srcdir}/build-${MSYSTEM}-shared"
+  mkdir -p "${srcdir}/build-${MSYSTEM}-shared" && cd "${srcdir}/build-${MSYSTEM}-shared"
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
```
Comment (jeremyd2019) [inline mingw-w64-minizip-ng/PKGBUILD:35]:
```suggestion
  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=;-DINSTALL_INC_DIR=" \
```
Something like this?

Comment (neheb) [reply mingw-w64-minizip-ng/PKGBUILD:35]: Yep. That worked.

**Comment by jeremyd2019** [comment]
what's the status of this?  It sounded like you were going to change the include dir

**Comment by neheb** [comment]
The situation is "interesting".

I looked at all packages that depend on minizip-git. A lot of them bundle their own version of minizip. It's by accident that they end up using the external one.

Also, nothing currently uses minizip-ng. I have no idea why it's even in the tree. I think the way forward is to make it replace minizp-git and patch all packages that don't compile.

**Comment by dennisameling** [comment]
> @dennisameling can you test this?

Not sure if this is still relevant, but I can confirm this builds correctly on arm64 👍🏼 

```
.....
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Stripping unneeded symbols from binaries and libraries...
  -> Compressing man and info pages...
==> Checking for packaging issues...
==> Creating package "mingw-w64-clang-aarch64-minizip-ng"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Finished making: mingw-w64-minizip-ng 3.0.4-1 (Mon Dec 27 15:48:31 2021)
```

**Comment by Neustradamus** [comment]
@neheb: Thanks!

Linked to:
- https://github.com/zlib-ng/minizip-ng/issues/358

**Comment by MehdiChinoune** [comment]
done in #14031

**Comment by Neustradamus** [comment]
@MehdiChinoune: Thanks for:
- https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-minizip-ng

@ all: It is possible to replace https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-minizip-git by https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-minizip with official zlib (https://github.com/madler/zlib)

Currently in 1.2.13 with CVE fixes?

**Comment by Neustradamus** [comment]
@MehdiChinoune: Thanks for:
- https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-minizip-ng

@ all: It is possible to replace https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-minizip-git by https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-minizip with official minizip from zlib (https://github.com/madler/zlib/tree/master/contrib/minizip)?

Currently in Zlib 1.2.13 with CVE fixes?

**Pull request closed without merging**